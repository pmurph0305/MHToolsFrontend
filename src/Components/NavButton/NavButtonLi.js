import React from "react";
import { Link } from "react-router-dom";
/**
 * @param {string} buttonLabel text displayed on the button
 * @param {string} liClasss className for each button li element default: null
 * @param {string} route Route for react router
 */
const NavButtonLi = ({ buttonLabel, route, liClass }) => {
  return (
    <li className={liClass ? liClass : null}>
      <Link to={`/${route}`}>
        <button>{buttonLabel}</button>
      </Link>
    </li>
  );
};

export default NavButtonLi;
