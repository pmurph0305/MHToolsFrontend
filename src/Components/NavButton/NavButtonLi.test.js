import { shallow, mount } from "enzyme";
import React from "react";
import NavButtonLi from "./NavButtonLi";
import { timeout } from "q";

describe("NavButtonLi tests", () => {
  let wrapper;
  let mockRouteChange = jest.fn(val => val);
  const mockProps = {
    buttonLabel: "test label",
    onRouteChange: mockRouteChange,
    route: "test route",
    liClass: "tc"
  };

  beforeEach(() => {
    wrapper = shallow(<NavButtonLi {...mockProps} />);
  });

  it("Exists", () => {
    expect(wrapper.exists()).toBe(true);
  });

  it("Matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Displays label correctly", () => {
    expect(wrapper.find("button").text()).toBe(mockProps.buttonLabel);
  });

  it("Has null class if not specified", () => {
    const noclasswrapper = shallow(<NavButtonLi />);
    expect(noclasswrapper.prop("className")).toBe(null);
  });

  // simulate a click, make sure correct route value is returned with click.
  it("On Click is called and sends route prop", () => {
    let wrapClick = shallow(<NavButtonLi {...mockProps} />);
    wrapClick.find("button").simulate("click");
    expect(mockRouteChange.mock.results[0].value).toBe("test route");
    expect(mockRouteChange).toHaveBeenCalledTimes(1);
  });
});
