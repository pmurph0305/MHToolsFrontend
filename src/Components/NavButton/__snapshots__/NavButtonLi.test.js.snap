// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavButtonLi tests Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavButtonLi
    buttonLabel="test label"
    liClass="tc"
    onRouteChange={[MockFunction]}
    route="test route"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        onClick={[Function]}
      >
        test label
      </button>,
      "className": "tc",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test label",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "test label",
      "type": "button",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          onClick={[Function]}
        >
          test label
        </button>,
        "className": "tc",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test label",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "test label",
        "type": "button",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
